### DISCUSSION TITLE
Questions about Noise Modelling possibilities

### DISCUSSION THREAD

**Discussion by AmandineHJ** [Universite-Gustave-Eiffel/NoiseModelling#564] [Q&A]
Hello everyone !

I have just begin using Noise Modelling in order to do research and tests for noise map. I have some questions about the possibilities of this tool, and how to apply it. I am using Noise Modelling version 4.0.4. 

I was wondering if it is possible to overlap noise sources ? For example : if I want to produce a noise map with the noise of the roads and two particular point that emit noise, is it possible to calculate the noise into the environment with these 3 types of sources at the same time ?  And if it is possible, how do I do that ?

I have another question. I have tried to change the type of receivers, and put a "regular grid" instead of "Delaunay grid", and then I tried to create isosurface. I couldn't create Isosurface because Noise Modelling was looking for the "TRIANGLES" table. So do I conclude that it is mandatory to use the "Delaunay Grid" when we want to create Isosurfaces ? Can I create isosurfaces or squares (like for example colored squares 5m x 5m) by linking the points to have a coloured visual rendered ?

I thank you in advance for your replies and have a nice day !

Amandine

<img width="957" alt="Capture _ 1" src="https://user-images.githubusercontent.com/129366571/228778091-8c12d568-25d1-4826-82be-dba7cf825839.png">
<img width="950" alt="Capture _ 2" src="https://user-images.githubusercontent.com/129366571/228778097-5fa9b763-f0f5-49b2-84b1-c4670b9c33b5.png">

**Comment by nicolas-f** [comment]
Hi,

After your feature request, I have added an option to generate Triangles in the RegularGrid WPS:

https://github.com/Universite-Gustave-Eiffel/NoiseModelling/pull/563

You can download the file https://github.com/Universite-Gustave-Eiffel/NoiseModelling/raw/326710fb9b0e68172fe644607a8342402b453716/wps_scripts/src/main/groovy/org/noise_planet/noisemodelling/wps/Receivers/Regular_Grid.groovy
and place it in the folder `data_dir/scripts/wps/Receivers/`

**Comment by AmandineHJ** [comment]
Hi,

Thank you very much for your reply. I have tried this new block by place the file into the good folder but it doesn't seems to work.  I have run the block "Regular Grid" with my arguments. It stopped running when deleting the receivers from the roads in the noise modelling window, and now it doesn't work anymore. You can see the error message in the picture, and I can't drop the table "TRIANGLES" anymore. It is frozen.

Furthermore, do you know more about my first question about overlap sources ?
Thank you in advance

<img width="952" alt="image" src="https://user-images.githubusercontent.com/129366571/228839438-affd3732-d2ee-4c5e-b8f7-ed2af5809e05.png">

**Comment by nicolas-f** [comment]
This error is when there is another sql query running in the background. You have to restart noisemodelling

In the noise propagation step Noise_level_from_source, the parameter **source table** is accepting POINT, MULTIPOINT, LINESTRING and MULTILINESTRING. So you can have roads(linestring) and point source (points) in the same table.

You should use geojson format in order to mix different type of geometries.

```
Name of the Sources table.

The table shall contain :
- PK : an identifier. It shall be a primary key (INTEGER, PRIMARY KEY).
- THE_GEOM : the 3D geometry of the sources (POINT, MULTIPOINT, LINESTRING, MULTILINESTRING). According to CNOSSOS-EU, you need to set a height of 0.05 m for a road traffic emission.
- LWD63, LWD125, LWD250, LWD500, LWD1000, LWD2000, LWD4000, LWD8000 : 8 columns giving the day emission sound level for each octave band (FLOAT).
- LWE* : 8 columns giving the evening emission sound level for each octave band (FLOAT).
- LWN* : 8 columns giving the night emission sound level for each octave band (FLOAT).
- YAW : Source horizontal orientation in degrees. For points 0° North, 90° East. For lines 0° line direction, 90° right of the line direction. (FLOAT).
- PITCH : Source vertical orientation in degrees. 0° front, 90° top, -90° bottom. (FLOAT).
- ROLL : Source roll in degrees (FLOAT).
- DIR_ID : identifier of the directivity sphere from tableSourceDirectivity parameter or train directivity if not provided -> OMNIDIRECTIONAL(0), ROLLING(1), TRACTIONA(2), TRACTIONB(3), AERODYNAMICA(4), AERODYNAMICB(5), BRIDGE(6) (INTEGER).
```

**Comment by AmandineHJ** [comment]
Thank you very much for you reply ! The creation of isosurfaces is very well working thanks to regular grid now. 
I have just on last question about this subject : I just saw in my tests that with the Delaunay grid, the sources (here the roads) appear in white (like you can see on the picture) and I prefer it to appear in the color that describes the noise level of the road (just like in the second picture, with the regular grid), is it possible to do that with the Delaunay grid ?

![image](https://user-images.githubusercontent.com/129366571/228876307-f4529a2c-5662-4c0d-ac6a-e7f4f875d70b.png)
![image](https://user-images.githubusercontent.com/129366571/228878404-90bd37d4-e488-42fe-9d61-65d06f91e2ca.png)

**Comment by nicolas-f** [comment]
Yes set **Source Width** parameter to -1

**Comment by AmandineHJ** [comment]
Thank you, I have just tried and it works very well ! Thank you for your time ! 
Just one last question about overlap sources : I didn't find any way of how I can combine different geometries with GeoJSON ... Could you help me ? For example : to test the tool, I am trying to take into account as a source a point, a polygon, and the roads. You can find attached my files. I tried many ways to combined the three files but I didn't succed in any method ... Thank you for your help

[Combine_sources.zip](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/files/11119789/Combine_sources.zip)

**Comment by nicolas-f** [comment]
You have to format lw_roads.json using an app like this

https://jsonformatter.curiousconcept.com

Then using your text editor (like sublime text for example) copy one feature node then replace the LineString geometry with a Point geometry. You can edit the spectrum of your new sound source to your liking.

```json
{
      "type":"Feature",
      "geometry":{
        "type":"LineString",
        "coordinates":[
          [
            249327.3,
            6746387.8,
            0.05
          ],
          [
            249340.2,
            6746360.7,
            0.05
          ],
          [
            249345.4,
            6746340.5,
            0.05
          ]
        ]
      }
"properties":{
        "PK":299,
        "LWD63":85.98
```


->
```json
{
      "type":"Feature",
      "geometry":{
        "type":"Point",
        "coordinates": [
            249327.3,
            6746387.8,
            0.05
          ]
      }
"properties":{
        "PK":300,
        "LWD63":90.00
```

You have to use the same projection system for all your coordinates.

You can use this tool (never used it) if you have valid geojson files for your other sources (must have the same properties than lw_source features)

https://findthatpostcode.uk/tools/merge-geojson

**Comment by AmandineHJ** [comment]
Thank you very much ! Combining the file has been done very well thanks to this tool : https://findthatpostcode.uk/tools/merge-geojson

I just have on problem, when I tried to import it on Noise Modelling, I had the following error message : 

![image](https://user-images.githubusercontent.com/129366571/229083023-c6ba59a3-5b52-456c-8011-ce0fd1873836.png)

I don't understand what it means?
Thanks for your help

**Reply by nicolas-f** [reply]
It means that you have multiple features with the same PK attribute, the primary key must be unique.

**Reply by pierromond** [reply]
you can just remove the PK column of your geojson files before merge them. NM will create a new unique PK column.

**Comment by AmandineHJ** [comment]
Hello, 
Thank you for your reply. I have tried to put my 3 sources (roads, one point, and one polygone), remove primary key and recreate a new PK column thanks to noise modelling but I have the following results :

![image](https://user-images.githubusercontent.com/129366571/229437512-caaef6bb-d63c-43fd-b81d-416f2de3a5fb.png)

I don't understand why my results are like this. I have checked my tables ( [Combine_sources.zip](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/files/11135745/Combine_sources.zip) ) but I see nothing special that could be a problem for Noise Modelling. The HEIGHT columns are not NULL and the primary key is present in all the table (after converting and putting the SOURCES_ESSAI document in geojson file and add a primary key via noise modelling). Do you have an idea ?

**Reply by AmandineHJ** [reply]
I have tried with only one point and the roads and in another try with only one polygon and the roads. It seems that the problem comes from the polygon. Maybe this is not possible to consider an entire polygon as a source ? Do you know something about that ?

**Reply by pierromond** [reply]
Yes indeed, surface sound sources are not taken into account in NM. You can trasnform you polygon in multiple point sound sources for example

**Reply by AmandineHJ** [reply]
Okay, thank you very much for your help !
